aptos-node/src/lib.rs

```diff
@@ -599,22 +604,51 @@ pub fn setup_environment_and_start_node(
         jwk_consensus_subscriptions,
     ) = state_sync::create_event_subscription_service(&node_config, &db_rw);
 
+    let peer_senders = Arc::new(OutboundPeerConnections::new());
+
     // Set up the networks and gather the application network handles
-    let peers_and_metadata = network::create_peers_and_metadata(&node_config);
-    let (
-        network_runtimes,
-        consensus_network_interfaces,
-        dkg_network_interfaces,
-        jwk_consensus_network_interfaces,
-        mempool_network_interfaces,
-        peer_monitoring_service_network_interfaces,
-        storage_service_network_interfaces,
-    ) = network::setup_networks_and_get_interfaces(
+    let peers_and_metadata = network2::create_peers_and_metadata(&node_config);
+    let (network_runtimes, mut networks) = network2::setup_networks(
         &node_config,
         chain_id,
         peers_and_metadata.clone(),
-        &mut event_subscription_service,
-    );
+        peer_senders.clone(),
+        &mut event_subscription_service);
+
+    let mut apps = ApplicationCollector::new();
+
+    // create map to lookup NetworkContext by NetworkId
+    let mut contexts = BTreeMap::new();
+    for network in networks.iter() {
+        let network_context = network.network_context();
+        contexts.insert(network_context.network_id(), network_context);
+    }
+    let contexts = Arc::new(contexts);
+
+    let app_setup = AppSetupContext{
+        node_config: node_config.clone(),
+        peers_and_metadata: peers_and_metadata.clone(),
+        peer_senders,
+        contexts,
+    };
+
+    let health_checker_network_interfaces = network2::health_checker_network_connections(&mut apps, &app_setup);
```

Comment:
I imagine you could roll up a lot of this new connectivity functionality into a separate PR? Even if we don't move to the "new interfaces" entirely, we could adopt some of the patterns in the existing code?

Reply:
Some of this refactor relies on the internal changes. I don't think it makes sense in the current code.